content/docs/fundamentals/converting/resolving-errors@ko.md

```diff
@@ -0,0 +1,350 @@
+
+이와 같은 요구사항으로 인해 다음에 명시된 경우를 제외하고는 AMP에서 모든 사용자 생성/타사 자바스크립트의 사용이 사실상 배제됩니다.
+
+{% call callout('참고', type='note') %}
+사용자 생성/타사 스크립트 사용 제한의 유일한 예외는 다음과 같습니다.
+
+1.  스크립트가 페이지에 메타데이터를 추가하거나 AMP 구성요소를 구성합니다. 이러한 스크립트에는 `application/ld+json` 또는 `application/json` 유형 속성이 포함됩니다.
+2.  스크립트가 iframes에 포함되어 있습니다.  iframe에 자바스크립트를 포함하는 일은 최후의 수단으로 생각해야 합니다. 가능하다면 자바스크립트 기능은 [AMP 구성요소](/ko/docs/reference/components.html)를 사용하여 대체해야 합니다. 다음 섹션에서 첫 번째 AMP 구성요소를 살펴보겠습니다.
+{% endcall %}
+
+외부 [`base.js`](https://github.com/googlecodelabs/accelerated-mobile-pages-foundations/blob/master/base.js) 파일을 열어 보세요. 무엇이 표시되나요? 이 파일에는 자바스크립트 코드가 없어야 하며 다음과 같은 정보의 주석만 포함되어 있어야 합니다.
+
+```javascript
+/*
+
+이 외부 자바스크립트 파일은 의도적으로 비어 있습니다.
+
+이 파일은 외부 자바스크립트 파일 사용과 관련된
+AMP 유효성 검사 오류를 보여주기 위해 사용됩니다.
+
+*/
+```
+
+이 외부 자바스크립트 파일은 웹사이트의 기능 구성요소가 아니므로 참조를 완전히 삭제할 수 있습니다.
+
+문서에서 다음 외부 자바스크립트 참조를 **삭제**합니다.
+
+```html
+<script type="text/javascript" src="base.js"></script>
+```
+
+이제 페이지를 **새로고침**하고 스크립트 오류가 사라졌는지 확인합니다.
+
+## AMP CSS 상용구 포함
+
+다음 오류는 누락된 상용구 코드를 참조합니다.
+
+<pre class="error-text">
+필수 'noscript enclosure for boilerplate' 태그가 누락되었거나 잘못되었습니다.
+필수 'head > style : boilerplate' 태그가 누락되었거나 잘못되었습니다.
+필수 'noscript > style : boilerplate' 태그가 누락되었거나 잘못되었습니다.
+</pre>
+
+모든 AMP 문서에는 다음과 같은 AMP 상용구 코드가 필요합니다.
+
+```html
+<style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+```
+
+문서의 `<head>` 태그 하단에 위의 상용구 코드를 **추가**합니다.
+
+`<style amp-boilerplate>` 태그를 사용하면 AMP 자바스크립트 라이브러리가 로드될 때까지 본문 콘텐츠가 숨겨지며, 그런 다음 콘텐츠가 렌더링됩니다. AMP에서는 스타일이 적용되지 않은 콘텐츠가 렌더링되는 일, 즉 FOUC(Flash Of Unstyled Content)를 방지하기 위해 이러한 작업을 수행합니다. 이를 통해 페이지의 콘텐츠가 한꺼번에 표시되며 스크롤 없이 볼 수 있는 부분이 한 번에 렌더링되므로 사용자 환경에서 진정한 의미의 즉각성을 경험할 수 있습니다. 두 번째 태그는 자바스크립트가 브라우저에서 사용 중지된 경우 이 논리를 되돌립니다.
+
+## `<img>`를 `<amp-img>`로 대체
+
+AMP에서는 미디어를 표시할 때 기본 HTML의 상응하는 페이지를 지원하지 않으므로 다음과 같은 오류가 발생합니다.
+
+<pre class="error-text">
+'img' 태그는 'noscript' 태그의 하위 요소로만 표시될 수 있습니다. 'amp-img'를 의미하셨나요?
+</pre>
+
+AMP에는 `<img>` 태그를 대체하기 위해 특별히 만들어진 웹 구성요소인 [`<amp-img>`](/ko/docs/reference/components/amp-img.html) 태그가 있습니다.
+
+```html
+<amp-img src="mountains.jpg"></amp-img>
+```
+
+`<img>` 태그를 위에 설명된 `<amp-img>` 태그로 **교체**하고 검사기를 다시 실행합니다. 다음과 같이 여러 가지 새로운 오류가 표시될 것입니다.
+
+<pre class="error-text">
```

Comment:
All of these code blocks need to be in EN, so please update with EN translation (https://www.ampproject.org/docs/fundamentals/converting/resolving-errors).